vendor/models/profile.py

```diff
@@ -111,11 +114,11 @@ def get_or_create_address(self, address):
         address, created = self.addresses.get_or_create(name=address.address_1, first_name=address.first_name, last_name=address.last_name, address_1=address.address_1, address_2=address.address_2, locality=address.locality, state=address.state, country=address.country, postal_code=address.postal_code, profile=self)
         return address, created
 
-    def get_customer_products(self):
+    def has_previously_owned_products(self, products):
         """
-        Get all products that a customer profile has ever owned
+        Get all products that the customer has purchased and returns True if it has.
         """
-        return set([ product for receipt in self.receipts.filter(status__gte=PurchaseStatus.COMPLETE) for product in receipt.products.all() ])
+        return bool(self.receipts.filter(Q(products__in=products), Q(status__gte=PurchaseStatus.COMPLETE)))
```

Comment:
Did you want to keep the previous method too?  It seems like it would be useful to get a list of user products.

Reply:
maybe something like this (to shorten the results returned and save on the DB call traffic):

```bool(self.receipts.filter(Q(products__in=products), Q(status__gte=PurchaseStatus.COMPLETE)).first())```

Reply:
I can't figure out how to return products from the receipts. as it is a MTM field the best thing I could think of was:
```
self.receipts.filter(status__gte=PurchaseStatus.COMPLETE).values_list('products', flat=True)
```
but that would only return products pks.

I do have a function below that returns all receipts that have been completed:
```
def get_completed_receipts(self):
        return self.receipts.filter(status__gte=PurchaseStatus.COMPLETE)
```